### ISSUE TITLE
Intalling quamash using PIP forces install of both pyside2 and pyqt5

### ISSUE THREAD

**Issue by ashleysommer** [harvimt/quamash#118]
Using latest code in the tests-2019 branch.
Installing via PIP using this entry in my requirements file:
`git+https://github.com/harvimt/quamash.git@tests-2019#egg=quamash`
PIP installs both PyQt5 and PySide2, where my project only uses PySide2.

I see in the pyproject file the requirements are marked as optional, but for some reason PIP installs them as required dependencies.

**Comment by ashleysommer** [comment]
Metadata generated by PIP after it downloads quamash but before installing it looks like this:
```
Metadata-Version: 2.1
Name: quamash
Version: 0.6.1
Summary: Implementation of the PEP 3156 Event-Loop with Qt.
Home-page: https://github.com/harvimt/quamash
License: BSD-2-Clause
Author: Mark Harviston
Author-email: mark.harviston@gmail.com> Arve Knudsen <arve.knudsen@gmail.com
Requires-Python: >=3.6,<4.0
Classifier: Development Status :: 3 - Alpha
Classifier: Environment :: X11 Applications :: Qt
Classifier: Intended Audience :: Developers
Classifier: License :: OSI Approved :: BSD License
Classifier: Operating System :: MacOS :: MacOS X
Classifier: Operating System :: Microsoft :: Windows
Classifier: Operating System :: POSIX
Classifier: Programming Language :: Python :: 3
Classifier: Programming Language :: Python :: 3.6
Classifier: Programming Language :: Python :: 3.7
Requires-Dist: PyQt5 (>=5.12,<6.0)
Requires-Dist: pyside2 (>=5.12,<6.0)
Project-URL: Repository, https://github.com/harvimt/quamash
Description-Content-Type: text/x-rst
```

Notice the two "Requires-Dist" entries.
Also notice it doesn't include any of the the items from "requirements.txt"

**Comment by ashleysommer** [comment]
Thought it might've been because I'm using slightly outdated pip and setuptools
Was on pip 9.0.3 and Setuptools 40.0.x
updated to pip 9.2.3 and setuptools 41.2.x
But still seeing the same problem.

**Comment by harvimt** [comment]
Yeah I've got no idea on this one. If you build a wheel file and then try
to install that?

**Comment by ashleysommer** [comment]
I'll open an issue on the PIP repo.

**Comment by ashleysommer** [comment]
Ok, tried cloning the repo, switching to tests-2019 branch and building the wheel.
Used `python3 setup.py bdist_wheel`
It creates a METADATA file with:
```
Metadata-Version: 2.1
Name: quamash
Version: 0.6.1
Summary: Implementation of the PEP 3156 Event-Loop with Qt.
Home-page: UNKNOWN
Author: Mark Harviston, Arve Knudsen
Author-email: mark.harviston@gmail.com, arve.knudsen@gmail.com
License: UNKNOWN
Platform: UNKNOWN
Classifier: Development Status :: 3 - Alpha
Classifier: Environment :: X11 Applications :: Qt
Classifier: Intended Audience :: Developers
Classifier: License :: OSI Approved :: BSD License
Classifier: Operating System :: MacOS :: MacOS X
Classifier: Operating System :: Microsoft :: Windows
Classifier: Operating System :: POSIX
Classifier: Programming Language :: Python :: 3
Classifier: Programming Language :: Python :: 3.6
Classifier: Programming Language :: Python :: 3.7
Description-Content-Type: text/x-rst
```
So it doesn't add those dependencies this time.
So looks like its definitely a problem with the way PIP generates the metadata file?

**Comment by chrahunt** [comment]
As mentioned over in pypa/pip#7127, I'd check with Poetry on this - it doesn't look like pip is doing anything strange here.

**Comment by ashleysommer** [comment]
Thank you @chrahunt for investigating this further
@harvimt Looks like this is a bug in the pyproject.toml file. I can write a PR for you that fixes it.

**Comment by harvimt** [comment]
That would be great.

**Comment by ashleysommer** [comment]
fixed via #119
